File: backend/app/models/quiz_attempt_question.py
from typing import TYPE_CHECKING
from sqlalchemy import Boolean, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base

if TYPE_CHECKING:
    from app.models.quiz_attempt import QuizAttempt


class QuizAttemptQuestion(Base):
    __tablename__ = "quiz_attempt_questions"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    quiz_attempt_id: Mapped[int] = mapped_column(
        ForeignKey("quiz_attempts.id", ondelete="CASCADE"),
        nullable=False,
        index=True
    )
    question_index: Mapped[int] = mapped_column(Integer, nullable=False)
    question_text: Mapped[str] = mapped_column(String(500), nullable=False)
    selected_answer: Mapped[int] = mapped_column(Integer, nullable=False)
    correct_answer: Mapped[int] = mapped_column(Integer, nullable=False)
    is_correct: Mapped[bool] = mapped_column(Boolean, nullable=False)
    topic: Mapped[str] = mapped_column(String(200), nullable=False)
    explanation: Mapped[str | None] = mapped_column(String(1000), nullable=True)

    quiz_attempt: Mapped["QuizAttempt"] = relationship(back_populates="questions")

File: backend/Models/QuizAttemptQuestion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VideoIntelligencePlatform.Backend.Models;

[Table("quiz_attempt_questions")]
public class QuizAttemptQuestion
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("quiz_attempt_id")]
    public int QuizAttemptId { get; set; }

    [Column("question_index")]
    public int QuestionIndex { get; set; }

    [Column("question_text")]
    [MaxLength(500)]
    public string QuestionText { get; set; } = string.Empty;

    [Column("selected_answer")]
    public int SelectedAnswer { get; set; }

    [Column("correct_answer")]
    public int CorrectAnswer { get; set; }

    [Column("is_correct")]
    public bool IsCorrect { get; set; }

    [Column("topic")]
    [MaxLength(200)]
    public string Topic { get; set; } = "General Concept";

    [Column("explanation")]
    [MaxLength(1000)]
    public string? Explanation { get; set; }

    [ForeignKey("QuizAttemptId")]
    public QuizAttempt? QuizAttempt { get; set; }
}
